meshtastic/tests/test_mesh_interface_behavioral_compat.py

```
@@ -0,0 +1,1849 @@
+
+@pytest.mark.unit
+class TestSendTelemetrySemanticDeprecation:
+    """Test sendTelemetry() semantic deprecation warnings."""
+
+    def test_sendTelemetry_unsupported_type_emits_warning(self, mock_interface, caplog):
+        """Verify unsupported telemetryType values emit deprecation warning."""
+        iface = mock_interface
+
+        with caplog.at_level(logging.WARNING):
+            # Send telemetry with unsupported type
+            with patch.object(iface, "_send_to_radio_impl"):
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)